Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/permission/CameraPermissionSettings.kt

```diff
@@ -0,0 +1,25 @@
+package de.rki.coronawarnapp.util.permission
+
+import android.content.Context
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.createFlowPreference
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CameraPermissionSettings @Inject constructor(
+    @AppContext private val context: Context
+) {
+    private val prefs by lazy {
+        context.getSharedPreferences("camera.permission.settings", Context.MODE_PRIVATE)
+    }
+
+    val isCameraDeniedPermanently = prefs.createFlowPreference(
+        key = PREFS_KEY_CAMERA_PERMISSION_DENIED,
+        defaultValue = false
+    )
```

Comment:
When is this set from `true` to `false`?

Can't we solve this without an extra preference? Just by checking `ActivityCompat.shouldShowRequestPermissionRationale`?

Reply:
Not with a combination with calling requestPermission in this screen which will trigger the system dialog earlier. Additionally this function returns false when user never asked for a permission and after denial . It is not helping in this case. If you have an idea let me know?

Reply:
It is not set from false again. In this case the user should have been done this from the device settings and I don't know any way to connect both. Still the user has actually denied it permanently before and as far as i know we can not show system the dialog anymore can we? . Everything will be reset only when user clears the data including the permissions

Reply:
> Still the user has actually denied it permanently before and as far as i know we can not show system the dialog anymore can we?

We can, after the user went to the app settings and changed it, that's the idea behind showing the camera permission card, isn't it?

>  Everything will be reset only when user clears the data including the permissions.

We need to check if we have camera permission or if `shouldShowRequestPermissionRationale` returns true,  and then reset the `isCameraDeniedPermanently` to false, would this work? :thinking: